tests/framework/utils.py

```diff
@@ -359,3 +365,44 @@ def eager_map(func, iterable):
     """Map version for Python 3.x which is eager and returns nothing."""
     for _ in map(func, iterable):
         continue
+
+
+def get_cpu_load(pid: int, iterations=20, omit=0):
+    """Get process threads CPU load."""
+    assert iterations > 0, "Get cpu load iterations must be at least 1."
+
+    if omit > 0:
+        time.sleep(omit)
+
+    cpu_load = dict()
+    for i in range(iterations):
+        cp = subprocess.run(GET_CPU_LOAD.format(pid), shell=True,
+                            check=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+
+        # Output contains all PID associated threads:
+        # 99.9 fc_vcpu 0
+        # 73.3 fc_vcpu 1
+        # 60.0 firecracker
+        # 0.0 fc_api
+        lines = cp.stdout.decode().split(sep="\n")
+
+        # This covers the case where the last output
+        # lines ends in `\n`, which results in an extra
+        # empty line.
+        if len(lines[-1]) == 0:
+            lines.pop()
+
+        if len(lines) == 0:
+            break
+
+        for line in lines:
+            info = line.strip().split(maxsplit=1, sep=" ")
+            if i == 0:
```

Comment:
I would test if `cpu_load[info[1]]` is not initialized. Otherwise there might be corner cases. For example if we get here before Firecracker fully started, the vcpu threads won't exist, so we won't initialized the list. And after a couple of iterations the vcpu threads will have started.

Reply:
Hmmm I don't see any change to this `if`. It's still 

```
if i == 0:
    cpu_load[info[1]] = list()
```

Reply:
Sorry, this comment somehow slipped. Please take a look again. The `get_cpu_percent` primitive is changed a bit.